### PR TITLE
Feature/conversation memory

### PR DESCRIPTION
**Pull request by collindutter**
- Moves docs on Conversation Memory Drivers to dedicated Drivers section of docs.
- Adds recipe for implementing "Sessions" with Amazon DynamoDb.

----
📚 Documentation preview 📚: https://griptape--751.org.readthedocs.build//751/

### CODE REVIEW

**Review by vachillo — DISMISSED** [review]

**Review by vachillo — DISMISSED** [review]

File: docs/examples/amazon-dynamodb-sessions.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+Griptape provides [Conversation Memory](../griptape-framework/structures/conversation-memory.md) as a means of persisting conversation context across multiple Structure runs.
+By default, Griptape's Conversation Memory does not provide a Coversation Memory Driver which means the conversation history is lost after the Structure is finished running.
```
Comment (SavagePencil) [inline docs/examples/amazon-dynamodb-sessions.md:2]: "If you provide it with a suitable a Driver, the memory of the previous conversation can be preserved between invocations (runs?) of a Structure, giving it additional context for how to respond."

**Review by SavagePencil — CHANGES_REQUESTED** [review]
spell out the benefit

File: docs/examples/amazon-dynamodb-sessions.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+Griptape provides [Conversation Memory](../griptape-framework/structures/conversation-memory.md) as a means of persisting conversation context across multiple Structure runs.
+If you provide it with a suitable a Driver, the memory of the previous conversation can be preserved between run of a Structure, giving it additional context for how to respond.
```
Comment (SavagePencil) [inline docs/examples/amazon-dynamodb-sessions.md:2]: "with a suitable a Driver" -> "with a suitable Driver"

(maybe were you playing Mario?)

**Review by SavagePencil — CHANGES_REQUESTED** [review]
mamma mia, the tiniest typo

File: docs/examples/amazon-dynamodb-sessions.md (hunk 2 above)
Comment (collindutter) [reply docs/examples/amazon-dynamodb-sessions.md]: It's-a-fixed

**Review by vachillo — APPROVED** [review]

**Review by SavagePencil — APPROVED** [review]

**Pull request merged**